ssh-audit.py

```diff
@@ -3713,14 +3720,55 @@ def audit(aconf: AuditConf, sshv: Optional[int] = None, print_target: bool = Fal
 
     return program_retval
 
+def algorithm_lookup(alg_names):       
+    if not alg_names:    
+        out.fail("[exception] Missing argument: alg_names")
+        sys.exit(1)
+        
+    alg_types = {'kex': 'key exchange algorithms', 
+                 'key': 'host-key algorithms',
+                 'mac': 'message authentication code algorithms',
+                 'enc': 'encryption algorithms (ciphers)'
+    }
+    
+    algorithms = alg_names.split(",")
+    adb = SSH2.KexDB.ALGORITHMS
+    algorithms_dict = {}
+    
+    for alg_type in alg_types:
+        for alg_name in algorithms:
+            try:
+                SSH2.KexDB.ALGORITHMS[alg_type][alg_name]
+                algorithms_dict.update( {alg_name : alg_type} ) 
+            except Exception:
```

Comment:
Instead of a blank `Exception` you should use a more specific one, else this obfuscates other errors and makes debugging harder. Here you could use a `KeyError`. 

Catching a bare exception is also called the "most diabolical antipattern" in Python.
https://realpython.com/the-most-diabolical-python-antipattern/

Reply:
Thanks for the feedback, I'm going to try and rewrite that logic entirely so there's no reliance on an error handler.

I'm thinking of something like this:

```python
for k_algtype, v in SSH2.KexDB.ALGORITHMS.items():
    for k_algname, v1 in v.items():
        if k_algname in algorithms:
            algorithms_dict.update( {k_algname : k_algtype} ) 
```